src/daemons/GraphDaemon.cpp

```diff
@@ -112,6 +101,19 @@ int main(int argc, char *argv[]) {
         LOG(WARNING) << "Number netio threads should be greater than zero";
         return EXIT_FAILURE;
     }
+
+    LOG(INFO) << "Starting Graph HTTP Service";
+    // http://127.0.0.1:XXXX/status is equivalent to http://127.0.0.1:XXXX
+    nebula::WebService::registerHandler("/status", [] {
+        return new nebula::graph::GraphHttpHandler();
+    });
+    status = nebula::WebService::start();
+    if (!status.ok()) {
+        LOG(ERROR) << "Failed to start web service: " << status;
+        nebula::WebService::stop();
```

Comment:
IMHO, This not a proper way to address this problem. `start` and `stop` ought to be paired only if the `start` succeeded. Otherwise, it's counter intuitive.

I think you should consider to join and destroy the internal thread if `start` fails.

Reply:
thanks very much, good point.